program/tests/namespace_royalty_enforcement.rs

```diff
@@ -0,0 +1,1123 @@
+        (
+            PayloadKey::Source.to_string(),
+            PayloadType::Pubkey(source.pubkey()),
+        ),
+        (
+            PayloadKey::Destination.to_string(),
+            PayloadType::Pubkey(dest.pubkey()),
+        ),
+    ]);
+
+    let transfer_wallet_to_wallet_operation = Operation::Transfer {
+        scenario: TransferScenario::WalletToWallet,
+    };
+
+    // Create a `validate` instruction.
+    let validate_ix = ValidateBuilder::new()
+        .rule_set_pda(rule_set_addr)
+        .mint(mint.pubkey())
+        .additional_rule_accounts(vec![
+            AccountMeta::new_readonly(source.pubkey(), false),
+            AccountMeta::new_readonly(dest.pubkey(), false),
+        ])
+        .build(ValidateArgs::V1 {
+            operation: transfer_wallet_to_wallet_operation.to_string(),
+            payload,
+            update_rule_state: false,
+            rule_set_revision: None,
+        })
+        .unwrap()
+        .instruction();
+
+    // Validate fail operation.
+    let err =
+        process_failing_validate_ix!(&mut context, validate_ix, vec![], Some(ADDITIONAL_COMPUTE))
+            .await;
+
+    // Check that error is what we expect.  The `IsWallet` rule currently returns `NotImplemented`.
+    match err {
+        solana_program_test::BanksClientError::TransactionError(
+            TransactionError::InstructionError(_, InstructionError::Custom(error)),
+        ) => {
+            assert_eq!(error, RuleSetError::NotImplemented as u32);
+        }
+        _ => panic!("Unexpected error: {:?}", err),
+    }
+}
+
+#[tokio::test]
+async fn wallet_to_prog_owned() {
+    let mut context = program_test().start_with_context().await;
+    let rule_set_addr = create_royalty_rule_set(&mut context).await;
+
+    // Create a Keypair to simulate a token mint address.
+    let mint = Keypair::new();
+
+    // Source key is a wallet.
+    let source = Keypair::new();
+
+    // Our destination key is going to be an account owned by the mpl-token-auth-rules program.
+    // Any one will do so for convenience we just use the RuleSet.
+
+    // Get on-chain account.
+    let on_chain_account = context
+        .banks_client
+        .get_account(rule_set_addr)
+        .await
+        .unwrap()
+        .unwrap();
+
+    // Account must have nonzero data to count as program-owned.
+    assert!(on_chain_account.data.iter().any(|&x| x != 0));
+
+    // Verify account ownership.
+    assert_eq!(mpl_token_auth_rules::ID, on_chain_account.owner);
+
+    let payload = Payload::from([
+        (PayloadKey::Amount.to_string(), PayloadType::Number(1)),
+        (
+            PayloadKey::Source.to_string(),
+            PayloadType::Pubkey(source.pubkey()),
+        ),
+        (
+            PayloadKey::Destination.to_string(),
+            PayloadType::Pubkey(rule_set_addr),
+        ),
+        (
+            PayloadKey::Authority.to_string(),
+            PayloadType::Pubkey(context.payer.pubkey()),
+        ),
```

Comment:
I think we are ok with sending payer pubkey for all of these since our tests don't really test for this case specifically.  I previously modified them all to send payer.pubkey, and then send it as `AccountMeta::new_readonly(context.payer.pubkey(), true),` to the additional rule accounts in the builder.

Could you just sanity check this one last time to make sure that's what we want to do?